state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -824,7 +838,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					egress_ids.push(egress_id);
 					transfer_params.push(TransferAssetParams {
 						asset,
-						amount,
+						amount: Self::withhold_transaction_fee(
+							IngressOrEgress::Egress,
+							asset,
+							amount,
+						),
 						to: destination_address,
 					});
 					DepositBalances::<T, I>::mutate(asset, |tracker| {
```

Comment:
Believe you should be passing the result from withhold_transaction_fee here.

Reply:
On reflection I'm not so sure - the fee is no longer in a deposit (ie. available to be spent), it's now in the withheld assets, which are earmarked for paying fees.

Reply:
I think if we do what alastair suggested, we also need to account for it in the other places where the deposit tracker is mutated. So basically the ```mark_as_fetched``` and the ```register_deposit``` functions should also take the net amount for accounting to be correct. Alternatively, we can leave it as as and include the fee in the accounting at all places DepositTracker is mutated.

Reply:
I think leaving it as is probably the best option. It would still do its job of tracking the deposits (kind off). Doing it the right way would require non-trivial changes.

Reply:
As mentioned in Discord - I think it's correct as is. The withheld assets should be considered 'spent' ie. they are no longer ours.